nir/src/main/scala/scala/scalanative/nir/Fresh.scala

```diff
@@ -3,13 +3,31 @@ package nir
 
 import java.util.concurrent.atomic.AtomicInteger
 
-final class Fresh(scope: String) {
-  private var i = new AtomicInteger(0)
-  def apply() = {
-    val value = i.getAndIncrement()
-    Local(scope, value)
+final class Fresh private (private var start: Int) {
+  def apply(): Local = {
+    start += 1
+    val value = start
+    Local(value)
```

Comment:
At the moment we don't guarantee binary compatibility of the tools, only of the libraries.